src/IO/WebSocket/WslinkImageStream/index.js

```diff
@@ -0,0 +1,160 @@
+      }, (unsubFailure) => {
+        console.log('Unsubscribe error ', unsubFailure);
+      });
+  };
+
+  // subscribeRenderTopic = () => {
+  //   model.client.VtkImageDelivery.onRenderChange(publicAPI.viewChanged).then((subscription) => {
+  //     model.renderTopicSubscription = subscription;
+  //   }, (subError) => {
+  //     console.log('Failed to subscribe to topic');
+  //     console.log(subError);
+  //   });
+  // }
+
+  publicAPI.removeRenderObserver = (viewId) => {
+    model.client.VtkImageDelivery.removeRenderObserver(viewId).then((successResult) => {
+      console.log(`Removed observer from view ${viewId} succeeded`);
+      console.log(successResult);
+    }, (failureResult) => {
+      console.log(`Failed to remove observer from view ${viewId}`);
+      console.log(failureResult);
+    });
+  };
+
+  // addRenderObserver(viewId) {
+  //   model.client.VtkImageDelivery.addRenderObserver(viewId).then((successResult) => {
+  //     console.log(`Successfully added observer to view ${viewId}`);
+  //     console.log(successResult);
+  //   }, (failureResult) => {
+  //     console.log(`Failed to add observer to view ${viewId}`);
+  //     console.log(failureResult);
+  //   });
+  // }
+
+  publicAPI.viewChanged = (data) => {
+    const msg = data[0];
+    const imgBlob = new Blob([msg.image], {
+      type: model.mimeType,
+    });
+    if (model.activeURL) {
+      window.URL.revokeObjectURL(model.activeURL);
+      model.activeURL = null;
+    }
+    model.activeURL = URL.createObjectURL(imgBlob);
+    const time = +(new Date());
+    model.fps = Math.floor(10000 / (time - model.lastTime)) / 10;
+    model.lastTime = time;
+
+    model.lastImageReadyEvent = {
+      url: model.activeURL,
+      fps: model.fps,
+      metadata: { size: msg.size, id: msg.id },
+    };
+
+    publicAPI.fireImageReady(model.lastImageReadyEvent);
+  };
+
+  /* eslint-disable camelcase */
+  publicAPI.connect = ({ view_id = -1, size = [500, 500] }) => (
+    // Subscribe to pubsub topic and add view observer
+    new Promise((resolve, reject) => {
+      model.view_id = view_id;
+      model.width = size[0];
+      model.height = size[1];
+
+      model.client.VtkImageDelivery.onRenderChange(publicAPI.viewChanged).then((subscription) => {
+        model.renderTopicSubscription = subscription;
+        model.client.VtkImageDelivery.addRenderObserver(view_id).then((successResult) => {
+          console.log(`Successfully added observer to view ${view_id}`);
+          console.log(successResult);
+          resolve(successResult);
+        }, (failureResult) => {
+          console.log(`Failed to add observer to view ${view_id}`);
+          console.log(failureResult);
+          reject(failureResult);
+        });
+      }, (subError) => {
+        console.log('Failed to subscribe to topic');
+        console.log(subError);
+        reject(subError);
+      });
+    })
+  );
+  /* eslint-enable camelcase */
+
+  publicAPI.destroy = () => {
```

Comment:
Don't forget about the stuff we did here.  Was that in my email from yesterday eve?